packages/element/src/create-interpolate-element.js

```diff
@@ -0,0 +1,326 @@
+			if ( stackDepth !== 0 ) {
+				const { stackLeadingText, tokenStart } = stack.pop();
+				output.push( indoc.substr( stackLeadingText, tokenStart ) );
+			}
+			addText();
+			return false;
+
+		case 'self-closed':
+			if ( 0 === stackDepth ) {
+				if ( null !== leadingTextStart ) {
+					output.push(
+						indoc.substr( leadingTextStart, startOffset - leadingTextStart )
+					);
+				}
+				output.push( elementCreator( conversionMap[ name ] )() );
+				offset = startOffset + tokenLength;
+				return true;
+			}
+
+			// otherwise we found an inner element
+			addChild(
+				new Component( elementCreator( conversionMap[ name ] ), [] ),
+				startOffset,
+				tokenLength
+			);
+			offset = startOffset + tokenLength;
+			return true;
+
+		case 'opener':
+			stack.push(
+				Frame(
+					new Component( elementCreator( conversionMap[ name ] ), [] ),
+					startOffset,
+					tokenLength,
+					startOffset + tokenLength,
+					leadingTextStart
+				)
+			);
+			offset = startOffset + tokenLength;
+			return true;
+
+		case 'closer':
+			// if we're not nesting then this is easy - close the block
+			if ( 1 === stackDepth ) {
+				addComponentFromStack( startOffset );
+				offset = startOffset + tokenLength;
+				return true;
+			}
+
+			// otherwise we're nested and we have to close out the current
+			// block and add it as a innerBlock to the parent
+			const stackTop = stack.pop();
+			const text = indoc.substr(
+				stackTop.prevOffset,
+				startOffset - stackTop.prevOffset
+			);
+			stackTop.component.children.push( text );
+			stackTop.prevOffset = startOffset + tokenLength;
+
+			addChild(
+				stackTop.component,
+				stackTop.tokenStart,
+				stackTop.tokenLength,
+				startOffset + tokenLength
+			);
+			offset = startOffset + tokenLength;
+			return true;
+
+		default:
+			addText();
+			return false;
+	}
+}
+
+/**
+ * Grabs the next token match in the string and returns it's details.
+ *
+ * @return  {Array}  An array of details for the token matched.
+ */
+function nextToken() {
+	const matches = tokenizer.exec( indoc );
+	// we have no more tokens
+	if ( null === matches ) {
+		return [ 'no-more-tokens' ];
+	}
+	const startedAt = matches.index;
+	const [ match, isClosing, name, isSelfClosed ] = matches;
+	const length = match.length;
+	if ( isSelfClosed ) {
+		return [ 'self-closed', name, startedAt, length ];
+	}
+	if ( isClosing ) {
+		return [ 'closer', name, startedAt, length ];
+	}
+	return [ 'opener', name, startedAt, length ];
+}
+
+/**
+ * Pushes text extracted from the indoc string to the output stack given the
+ * current rawLength value and offset (if rawLength is provided ) or the
+ * indoc.length and offset.
+ *
+ * @param   {number}  rawLength  If provided will be used as the length of chars
+ *                               to extract.
+ */
+function addText( rawLength ) {
+	const length = rawLength ? rawLength : indoc.length - offset;
+	if ( 0 === length ) {
+		return output.push( '' );
+	}
```

Comment:
If we stop pushing `''` to the output we might be able to get rid of the `filter( a => '' !== a )` in the output

so we'd do less and be faster doing it

Reply:
see d980d8f